src/pages/ShoppingList/ShoppingListDetails.jsx

```diff
@@ -1,38 +1,105 @@
-import React from "react";
+import React, { useState } from "react";
+import classNames from "classnames";
 import { useParams } from "react-router-dom";
-import { Checkbox } from "@/components";
+import { Checkbox, Text, Button, Card } from "@/components";
+import styles from "./ShoppingList.module.scss";
+import useShoppingListDetails from "./hooks/api/useShoppingListDetails";
+import useMarkShoppingListDetail from "./hooks/api/useMarkShoppingListDetail";
 
-function ShoPpingListDetails() {
+function ShoppingListDetails() {
+  const [checked, setChecked] = useState(new Array(4).fill(false));
+  const { mutate } = useMarkShoppingListDetail();
```

Comment:
Wracając do naszej ostatniej rozkminy - co jeśli zamiast jednego hooka, mielibyśmy hook per checkbox? Czy to rozwiązałoby nasz problem, gdy `onSuccess` dla mutate wywołuje się tylko raz? Moglibyśmy wyciągnąć ten hook do oddzielnego komponentu, który renderuje element listy, obsługuje klikanie na checkbox i wówczas modyfikować optymistycznie dane z cache'a dla danych z `useShoppingListDetails`. Co myślisz o takim pomyśle na rozwiązanie tego problemu?

Reply:
Mhmm, w sumie o tym nie pomyślałem - a jak wtedy obsługiwać hook tego elementu listy z poziomu page'a?

Reply:
Niech dostanie przez propsy stan dla konkretnego elementu listy i... tyle? Sam w sobie niech ma hooka od update'a, który będzie optymistycznie aktualizował stan cache'a, który będzie aktualizowany na page'u, który będzie przekazywał dane dalej do elementu listy (poprawnie ustawiając jego stan). Dodatkowo możesz disablować checkboxy niezależnie od siebie na czas trwania requestu, który robi update

Reply:
Oki, rozumiem, ma to sens 😅 Postaram się w takim razie to zaktualizować przed następnym wtorkiem :)

Reply:
To na spokojnie, wylądujmy co mamy teraz, żeby już można było klikać po checkboxach, a potem możemy pogłówkować nad tą mechaniką i zrobić dla niej improvementy 😉